taxi/views.py

```diff
@@ -16,22 +17,41 @@ def index(request):
 
     num_visits = request.session.get("num_visits", 0)
     request.session["num_visits"] = num_visits + 1
-
     context = {
         "num_drivers": num_drivers,
         "num_cars": num_cars,
         "num_manufacturers": num_manufacturers,
         "num_visits": num_visits + 1,
     }
-
     return render(request, "taxi/index.html", context=context)
 
 
-class ManufacturerListView(LoginRequiredMixin, generic.ListView):
+class PreviousURLMixin:
```

Comment:
Please get rid this mixin and find another solution

Reply:
Ok, I'll rollback to the previous solution, but what wrong in this mixin?